tt_metal/llrt/tt_cluster.cpp

```diff
@@ -556,26 +546,26 @@ void Cluster::dram_barrier(chip_id_t chip_id) const {
 // binaries, metadata, and data to compute on are committed before launching kernels
 void Cluster::l1_barrier(chip_id_t chip_id) const {
     // Sets and resets L1 barrier of all tensix cores and ethernet cores
-    this->get_driver(chip_id).l1_membar(chip_id, "LARGE_WRITE_TLB");
+    this->driver_->l1_membar(chip_id, "LARGE_WRITE_TLB");
 }
 
 uint32_t Cluster::get_num_host_channels(chip_id_t device_id) const {
     bool mmio_capable = this->cluster_desc_->is_chip_mmio_capable(device_id);
-    return mmio_capable ? this->get_driver(device_id).get_num_host_channels(device_id) : 0;
+    return mmio_capable ? this->driver_->get_num_host_channels(device_id) : 0;
 }
 
 uint32_t Cluster::get_host_channel_size(chip_id_t device_id, uint32_t channel) const {
     TT_ASSERT(this->cluster_desc_->is_chip_mmio_capable(device_id));
-    return this->get_driver(device_id).get_host_channel_size(device_id, channel & HOST_MEM_CHANNELS_MASK);
+    return this->driver_->get_host_channel_size(device_id, channel & HOST_MEM_CHANNELS_MASK);
 }
 
 void *Cluster::host_dma_address(uint64_t offset, chip_id_t src_device_id, uint16_t channel) const {
     TT_ASSERT(this->cluster_desc_->is_chip_mmio_capable(src_device_id));
-    return this->get_driver(src_device_id).host_dma_address(offset, src_device_id, channel & HOST_MEM_CHANNELS_MASK);
+    return this->driver_->host_dma_address(offset, src_device_id, channel & HOST_MEM_CHANNELS_MASK);
 }
 
 uint64_t Cluster::get_pcie_base_addr_from_device(chip_id_t chip_id) const {
-    return this->get_driver(chip_id).get_pcie_base_addr_from_device();
+    return this->driver_->get_pcie_base_addr_from_device();
```

Comment:
since this does not rely on `chip_id` anymore could you remove the parameter

Reply:
Even though this would be easier at the moment... It makes more sense that this API does depend on chip_id. I'll make a change in UMD to reflect this.

Reply:
Added another UMD PR which addresses this, and modified the code here to pass chip_id.